docs/fundamentals/interface.rst

```diff
@@ -0,0 +1,105 @@
+
+- **Dark gray**: Awaiting initialization
+- **Light gray**: Awaiting execution
+- **Yellow**: Currently executing
+- **Green**: Successfully finished execution
+- **Red**: Execution failed
+
+.. figure:: execution_finished.png
+    :align: center
+    :width: 50%
+
+Resources
+---------
+.. figure:: resources_tab.png
+    :align: center
+    :width: 75%
+
+This view shows the available neural network resources available to your pipelines. From this view, resources can be created, deleted, and renamed. Resources are arranged by the deepforge system and cannot by manually reordered.
+
+.. figure:: neural_network.png
+    :align: center
+    :width: 50%
+
+As with pipelines, the neural networks are depicted as directed graphs. Each node in the graph corresponds to a single layer or operation in the network (information on operations can be found on the `keras website <https://keras.io/api/>`_). Clicking on a node provides the ability to change the attributes of that layer, delete the layer, or add new layers before or after the current node. Many operations require that certain attributes be defined before use. The Conv2D node pictured above, for example, requires that the *filters* and *kernel_size* attributes be defined. If these are left as *<none>*, a visual indicator will show that there is an error to help prevent mistakes. In order to ease analysis and development, hovering over any connecting line will display the shape of the data as it moves between the given layers.
+
+Artifacts
+---------
+.. figure:: artifacts_tab.png
+    :align: center
+    :width: 75%
+
+In this view, you can see all artifacts that are available to your pipelines. These artifacts can be used in any pipeline through the inclusion of the built in **Input** node. Artifacts are, by default, only supported in the form of either keras models (such as those created using the `keras.model.save_model <https://keras.io/api/models/model_saving_apis/#save_model-function>`_ function) or python pickle objects. Other artifact types can also be used, but require the definition of a `custom serialization <Custom Serialization_>`_. A new artifact can be created in one of three ways. First, artifacts are automatically during the execution of any pipeline that includes the built-in **Output** node. Second, artifacts can be directly uploaded in this view using the red upload button in the bottom right of the workspace. Using this option will also upload the artifact to the storage platform specified in the popup window. Finally, artifacts that already exist in one of the storage platforms can be imported using the blue import button in the bottom right of the workspace.
```

Comment:
> Artifacts are, by default, only supported in the form of either keras models (such as those created using the `keras.model.save_model <https://keras.io/api/models/model_saving_apis/#save_model-function>`_ function) or python pickle objects.

I am not sure if this is entirely accurate (maybe I am misunderstanding what you mean). Artifacts are just some arbitrary data with metadata about the corresponding Python type. The type information is used to deserialize the data. If the type does not have a custom serializer defined (or is omitted), then pickle is used. Otherwise, it uses the defined deserializer.

Reply:
You are correct that my original description is unclear and not quite correct unless discussing deepforge-keras specifically. I have changed
```Artifacts are, by default, only supported in the form of either keras models (such as those created using the keras.model.save_model function) or python pickle objects. Other artifact types can also be used, but require the definition of a custom serialization.```
to the following.
```Artifacts are pieces of saved data that may be associated with some Python data type. Any arbitrary type of data may be used for creating an artifact, but If a data type is not specified, or if a data type is not provided with a `custom serialization <Custom Serialization_>`_, the artifact will be treated as a `pickle object <https://docs.python.org/3/library/pickle.html>`_. If you have data that cannot be opened with Python's pickle module, you will need to create a custom serialization as described below. Some deepforge extenstions may also support additional data types by default. Deepforge-keras, for example, supports saved keras models, in addition to the standard pickle objects, without the need for custome serializtion.```

I hope that this is more clear and correct.

Reply:
There are two minor typos in "custom serialization" at the end but otherwise it sounds great!